### PR TITLE
Adds supermatter golems

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

Adds supermatter golems. They can be created only through magic mirror, staff of change or badminery. Basicly, they are very dangerous glass cannons that can do shit, but get killed by two shots of a laser cannon.

Downsides:
-Take x1,5 brute damage
-Take x3 burn damage
-Can’t pick up anything
-Aren’t heat immune like other golems
-Don’t have their own sprite because i am not a spriter
-Explode on death
-Have a chance to explode every second they are in crit

Benefits:
-They arms are literally supermatter swords
-They slowly regenerate damage
-They can’t be hurt by ballistic weapons
-No hardcrit and softcrit
-Explode when killed!

# Wiki Documentation

Add this guy to golem subtypes

# Changelog

:cl:  
rscadd: Adds supermatter golems  
/:cl:

### CODE REVIEW

**Comment by neeshacark** [comment]
I hope that pride ruin doesn't allow you to become a sm golem?

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 1]
```diff
@@ -1464,3 +1464,84 @@
 	wax_golem.visible_message(span_danger("[src] rises and reforms into [wax_golem]!"),span_userdanger("You reform into yourself!"))
 	wax_golem = null
 	qdel(src)
+
+//Teleports when hit or when it wants to
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 2]
```diff
@@ -1464,3 +1464,83 @@
...
+		H.visible_message(span_danger("[P] melts on collision with [H]!"))
+		playsound(get_turf(src), 'sound/effects/supermatter.ogg', 10, TRUE)
+		return BULLET_ACT_FORCE_PIERCE 
+
+/datum/species/golem/supermater/spec_life(mob/living/carbon/C)
+	. = ..()
+	C.adjustBurnLoss(-src.burnheal)
+	C.adjustBruteLoss(-src.bruteheal)
+	if(C.InCritical() && prob(1.5))
+		randexplode = TRUE
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
		src.randexplode = TRUE
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 3]
```diff
@@ -1464,3 +1464,83 @@
...
+		visible_message(span_danger("[C]'s body violently explodes!"))
+		explosion(get_turf(H), 1 ,3, 6)
+		qdel(C)
+
+/datum/species/golem/supermater/spec_death(gibbed, mob/living/carbon/human/H)
+	if(gibbed)
+		return
+	if(randexplode) //No double explosions
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	if(src.randexplode) //No double explosions
```

**Comment by ghost** [comment]
> I hope that pride ruin doesn't allow you to become a sm golem?

No, it doesn’t

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 4]
```diff
@@ -1464,3 +1464,83 @@
...
+		playsound(get_turf(H), 'sound/effects/supermatter.ogg', 10, TRUE)
+		visible_message(span_danger("[I] knocks into [target], and then turns into dust in a flash of light!"))
+		qdel(I)
+	
+
+/datum/species/golem/supermater/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style)
+	..()
+	visible_message(span_danger("[M] puches [H], but then turns into dust in a brilliant flash of light!"))
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 10, TRUE)
+	H.dust
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	H.dust()
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 5]
```diff
@@ -1464,3 +1464,83 @@
...
+		return BULLET_ACT_FORCE_PIERCE 
+
+/datum/species/golem/supermater/spec_life(mob/living/carbon/C)
+	. = ..()
+	C.adjustBurnLoss(-src.burnheal)
+	C.adjustBruteLoss(-src.bruteheal)
+	if(C.InCritical() && prob(1.5))
+		randexplode = TRUE
+		visible_message(span_danger("[C]'s body violently explodes!"))
+		explosion(get_turf(H), 1 ,3, 6)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
		explosion(get_turf(C), 1 ,3, 6)
```

**Comment by ghost** [comment]
Fuck i again accidentally clicked review button

**Review by ghost — COMMENTED** [review]
E

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 6]
```diff
@@ -1464,3 +1464,83 @@
...
+	attack_verb = "bluespace punch"
+	attack_sound = 'sound/effects/supermatter.ogg'
+	fixed_mut_color = "ff0"
+	brutemod = 1.5
+	burnmod = 3
+	var/burnheal = 1
+	var/bruteheal = 0.5
+	var/randexplode = FALSE
+
+/datum/species/golem/supermater/spec_hitby(atom/movable/AM, mob/living/carbon/human/H)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
/datum/species/golem/supermatter/spec_hitby(atom/movable/AM, mob/living/carbon/human/H)
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 7]
```diff
@@ -1464,3 +1464,83 @@
...
+	var/obj/item/I
+	I = AM
+	if(I)
+		H.AdjustFireLoss(2)
+		playsound(get_turf(H), 'sound/effects/supermatter.ogg', 10, TRUE)
+		visible_message(span_danger("[I] knocks into [target], and then turns into dust in a flash of light!"))
+		qdel(I)
+	
+
+/datum/species/golem/supermater/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
/datum/species/golem/supermatter/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style)
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 8]
```diff
@@ -1464,3 +1464,83 @@
...
+	..()
+	visible_message(span_danger("[M] puches [H], but then turns into dust in a brilliant flash of light!"))
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 10, TRUE)
+	H.dust()
+
+/datum/species/golem/supermater/spec_attacked_by(obj/item/I, mob/living/user, obj/item/bodypart/affecting, intent, mob/living/carbon/human/H)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
/datum/species/golem/supermatter/spec_attacked_by(obj/item/I, mob/living/user, obj/item/bodypart/affecting, intent, mob/living/carbon/human/H)
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 9]
```diff
@@ -1464,3 +1464,83 @@
...
+	..()
+	visible_message(span_danger("[user] attacks [H] with [I], and then [I] turns into dust in a flash of light!"))
+	playsound(get_turf(H), 'sound/effects/supermatter.ogg', 10, TRUE)
+	qdel(I)
+	
+
+/datum/species/golem/supermater/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
/datum/species/golem/supermatter/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 10]
```diff
@@ -1464,3 +1464,83 @@
...
+	..()
+	if(!istype(P, /obj/item/projectile/ion/) && !istype(P, /obj/item/ammo_casing/energy/laser/) && !istype(P, /obj/item/projectile/magic))
+		H.visible_message(span_danger("[P] melts on collision with [H]!"))
+		playsound(get_turf(src), 'sound/effects/supermatter.ogg', 10, TRUE)
+		return BULLET_ACT_FORCE_PIERCE 
+
+/datum/species/golem/supermater/spec_life(mob/living/carbon/C)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
/datum/species/golem/supermatter/spec_life(mob/living/carbon/C)
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 11]
```diff
@@ -1464,3 +1464,83 @@
...
+	. = ..()
+	C.adjustBurnLoss(-src.burnheal)
+	C.adjustBruteLoss(-src.bruteheal)
+	if(C.InCritical() && prob(1.5))
+		src.randexplode = TRUE
+		visible_message(span_danger("[C]'s body violently explodes!"))
+		explosion(get_turf(C), 1 ,3, 6)
+		qdel(C)
+
+/datum/species/golem/supermater/spec_death(gibbed, mob/living/carbon/human/H)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
/datum/species/golem/supermatter/spec_death(gibbed, mob/living/carbon/human/H)
```

**Review by ghost — COMMENTED** [review]
Fucking github

**Comment by NCRLouTenant** [comment]
If this doesn't have it's own sprite, how are you supposed to know a golem is a SM golem?

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 12]
```diff
@@ -1464,3 +1464,83 @@
...
+	var/randexplode = FALSE
+
+/datum/species/golem/supermatter/spec_hitby(atom/movable/AM, mob/living/carbon/human/H)
+	..()
+	var/obj/item/I
+	I = AM
+	if(I)
+		H.AdjustFireLoss(2)
+		playsound(get_turf(H), 'sound/effects/supermatter.ogg', 10, TRUE)
+		visible_message(span_danger("[I] knocks into [target], and then turns into dust in a flash of light!"))
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
		H.visible_message(span_danger("[I] knocks into [target], and then turns into dust in a flash of light!"))
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 13]
```diff
@@ -1464,3 +1464,83 @@
...
+		qdel(I)
+	
+
+/datum/species/golem/supermatter/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style)
+	..()
+	visible_message(span_danger("[M] puches [H], but then turns into dust in a brilliant flash of light!"))
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	H.visible_message(span_danger("[M] puches [H], but then turns into dust in a brilliant flash of light!"))
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 14]
```diff
@@ -1464,3 +1464,83 @@
...
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 10, TRUE)
+	H.dust()
+
+/datum/species/golem/supermatter/spec_attacked_by(obj/item/I, mob/living/user, obj/item/bodypart/affecting, intent, mob/living/carbon/human/H)
+	..()
+	visible_message(span_danger("[user] attacks [H] with [I], and then [I] turns into dust in a flash of light!"))
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	H.visible_message(span_danger("[user] attacks [H] with [I], and then [I] turns into dust in a flash of light!"))
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 15]
```diff
@@ -1464,3 +1464,83 @@
...
+		playsound(get_turf(src), 'sound/effects/supermatter.ogg', 10, TRUE)
+		return BULLET_ACT_FORCE_PIERCE 
+
+/datum/species/golem/supermatter/spec_life(mob/living/carbon/C)
+	. = ..()
+	C.adjustBurnLoss(-src.burnheal)
+	C.adjustBruteLoss(-src.bruteheal)
+	if(C.InCritical() && prob(1.5))
+		src.randexplode = TRUE
+		visible_message(span_danger("[C]'s body violently explodes!"))
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
		C.visible_message(span_danger("[C]'s body violently explodes!"))
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 16]
```diff
@@ -1464,3 +1464,83 @@
...
+		explosion(get_turf(C), 1 ,3, 6)
+		qdel(C)
+
+/datum/species/golem/supermatter/spec_death(gibbed, mob/living/carbon/human/H)
+	if(gibbed)
+		return
+	if(src.randexplode) //No double explosions
+		return
+	visible_message(span_danger("[H]'s body violently explodes!"))
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	H.visible_message(span_danger("[H]'s body violently explodes!"))
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 17]
```diff
@@ -1464,3 +1464,83 @@
...
+	explosion(get_turf(H), 1 ,3, 6)
+	qdel(C)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	qdel(H)
```

**Comment by ghost** [comment]
> If this doesn't have it's own sprite, how are you supposed to know a golem is a SM golem?

It has burning hands. Also that thing that it doesn't have its own sprite is not intended, i am just a VERY BAD spriter so I can't make it myself.

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 18]
```diff
@@ -1464,3 +1464,83 @@
...
+	var/randexplode = FALSE
+
+/datum/species/golem/supermatter/spec_hitby(atom/movable/AM, mob/living/carbon/human/H)
+	..()
+	var/obj/item/I
+	I = AM
+	if(I)
+		H.AdjustFireLoss(2)
+		playsound(get_turf(H), 'sound/effects/supermatter.ogg', 10, TRUE)
+		H.visible_message(span_danger("[I] knocks into [target], and then turns into dust in a flash of light!"))
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
		H.visible_message(span_danger("[I] knocks into [H], and then turns into dust in a flash of light!"))
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 19]
```diff
@@ -1464,3 +1464,83 @@
...
+	var/burnheal = 1
+	var/bruteheal = 0.5
+	var/randexplode = FALSE
+
+/datum/species/golem/supermatter/spec_hitby(atom/movable/AM, mob/living/carbon/human/H)
+	..()
+	var/obj/item/I
+	I = AM
+	if(I)
+		H.AdjustFireLoss(2)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
		H.adjustFireLoss(2)
```

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 20]
```diff
@@ -1464,3 +1464,83 @@
...
+/datum/species/golem/supermatter/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	..()
+	if(!istype(P, /obj/item/projectile/ion/) && !istype(P, /obj/item/ammo_casing/energy/laser/) && !istype(P, /obj/item/projectile/magic))
+		H.visible_message(span_danger("[P] melts on collision with [H]!"))
+		playsound(get_turf(src), 'sound/effects/supermatter.ogg', 10, TRUE)
+		return BULLET_ACT_FORCE_PIERCE 
+
+/datum/species/golem/supermatter/spec_life(mob/living/carbon/C)
+	. = ..()
+	C.adjustBurnLoss(-src.burnheal)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	C.adjustFireLoss(-src.burnheal)
```

**Comment by ynot01** [comment]
i don't think staff of change can make you different races, making this a solely wizard race... i think this is funny but it should be more accessible, perhaps able to make SM golem with a harvested SM shard?

**Comment by ghost** [comment]
> i don't think staff of change can make you different races, making this a solely wizard race... i think this is funny but it should be more accessible, perhaps able to make SM golem with a harvested SM shard?

Hmmmm... I will try to make this later

**Comment by JamieD1** [comment]
I honestly dont like the idea of someone running round able to dust almost anything instantly.

**Comment by ghost** [comment]
> I honestly dont like the idea of someone running round able to dust almost anything instantly.

I can make it admin only if necessary

**Comment by JamieD1** [comment]
> > I honestly dont like the idea of someone running round able to dust almost anything instantly.
> 
> I can make it admin only if necessary

The issue is dusting is irreversible, can you make it admin only for now

**Comment by ghost** [comment]
> > > I honestly dont like the idea of someone running round able to dust almost anything instantly.
> 
> > 
> 
> > I can make it admin only if necessary
> 
> 
> 
> The issue is dusting is irreversible, can you make it admin only for now

Yes

**Comment by ghost** [comment]
> The issue is dusting is irreversible, can you make it admin only for now

Done.

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**